snapfaas/src/fs/utils.rs

```diff
@@ -0,0 +1,129 @@
+use super::*;
+use labeled::buckle::{Buckle, Component};
+use labeled::Label;
+
+pub fn create_or_update_blob<S: BackingStore, P: Into<self::path::Path>>(
+    fs: &FS<S>,
+    base_dir: P,
+    name: String,
+    label: Buckle,
+    blob_name: String,
+) -> Result<(), FsError> {
+    if let DirEntry::Directory(dir) = fs.read_path(base_dir)? {
+        match dir.list(fs).get(&name) {
+            Some(DirEntry::Blob(blobentry)) => {
+                blobentry.replace(blob_name, fs).map_err(Into::into)
+            },
+            Some(_) => {
```

Comment:
So these util functions are basically _only_ used for bootstrap and adminfs_tools, and are thus specialized for those cases (they only live here for historical reasons).

In the case of bootstrapping, we basically want to replace whatever was there before with a blob since we're trying to ensure the filesystem is in  a particular state. This is not the right behavior in other cases.

Reply:
got it